up-l1/README.adoc

```diff
@@ -122,9 +122,21 @@ On the other hand, unsuccessful completion of this method does not necessarily m
 
 NOTE: The above strategy for retrying failed attempts to send a message results in https://www.cloudcomputingpatterns.org/at_least_once_delivery/[at-least-once delivery]. Recipient(s) of these messages should therefore be https://www.cloudcomputingpatterns.org/idempotent_processor/[Idempotent Processors].
 
-UTransport implementations
 
+[.specitem,oft-sid="des~utransport-send-preserve-data~1",oft-needs="impl,utest"]
+--
 * *MUST* preserve all of the message's meta data and payload during transmission
+--
+
+[.specitem,oft-sid="req~utransport-send-error-invalid-parameter~1",oft-needs="impl,utest"]
```

Comment:
I would also place this in the _design_ realm ...

```suggestion
[.specitem,oft-sid="dsn~utransport-send-error-invalid-parameter~1",oft-needs="impl,utest"]
```

Reply:
Arn't error values returned user requirements for the API and not design/architecture considerations? that is why I made some `req` and others `dsn`

Reply:
IMHO something like
> In case of an error, the API must provide specific information to client code regarding the reason of the error.

would be a (user) requirement. If we already explicitly define the error codes and under which circumstances we want to return them, then IMHO we have already made a design decision as to how the API looks like. Which is a reasonable thing to do here because we want to make sure that every language library implements this the same way.